src/plugin/adapter-registry/c-tracker.ts

```diff
@@ -58,16 +58,45 @@ export class CTracker extends AdapterVariableTracker {
         }
         this.logger.debug('Resolved', variable.name, { start: variableAddress, size: variableSize });
         const address = BigInt(variableAddress);
+        const startAddress = toHexStringWithRadixMarker(address);
+        const isPointer = this.isMaybePointer(variable, startAddress);
         const variableRange: VariableRange = {
             name: variable.name,
-            startAddress: toHexStringWithRadixMarker(address),
+            startAddress,
             endAddress: variableSize === undefined ? undefined : toHexStringWithRadixMarker(address + variableSize),
             value: variable.value,
             type: variable.type,
+            isPointer,
         };
         return variableRange;
     }
 
+    override async getLocals(session: vscode.DebugSession): Promise<VariableRange[]> {
+        this.getAddressSpaces(session);
+        return super.getLocals(session);
+    }
+
+    async getAddressSpaces(session: vscode.DebugSession): Promise<void> {
+        // The REPL output is not returned, only sent as a message & captured by onDidSendMessage below.
+        return session.customRequest('evaluate', { expression: '> info proc mappings', context: 'repl', frameId: this.currentFrame });
```

Comment:
This looks fairly GDB-specific, but at the moment, this tracker is also used for the ARM debuggers. We can:
 - Add a GDB-specific tracker to be used only for GDB.
 - Trust users only to go try to 'go to' values that really are addresses, and let them run into error handlers otherwise.